docs2/site/docs/migrations/migration4.md

```diff
@@ -2,105 +2,291 @@
+value set on the field, of course) and converts the name to uppercase:
+
+```cs
+public class HumanInputType : InputObjectGraphType
+{
+    public HumanInputType()
+    {
+        Name = "HumanInput";
+        Field<NonNullGraphType<StringGraphType>>("name");
+        Field<StringGraphType>("homePlanet");
+    }
+
+    public override object ParseDictionary(IDictionary<string, object> value)
+    {
+        return new Human
+        {
+            Name = ((string)value["name"]).ToUpper(),
+            HomePlanet = value.TryGetValue("homePlanet", out var homePlanet) ? (string)homePlanet : "Unknown",
+            Id = null,
+        };
+    }
+}
+```
+
+Note that pursuant to GraphQL specifications, if a field is optional, not supplied, and has no default,
+it will not be in the dictionary.
+
+For untyped `InputObjectGraphType` classes, like shown above, the default behavior of `ParseDictionary`
+will be to return the dictionary. `GetArgument<T>` will still attempt to convert a dictionary to the
+requested type via `ObjectExtensions.ToObject` as it did before.
+
+### Experimental Features / Applied Directives
+
+> Ability to apply directives to the schema elements and expose user-defined meta-information
+  via introspection - `schema.EnableExperimentalIntrospectionFeatures()`.
+> See https://github.com/graphql/graphql-spec/issues/300 for more information.
+
+(sungam3r todo: write separate page on these features, write a simple summary and reasoning here, and add link to new page)
+
+### Ability to Sort Introspection Results
+
+Introspection results are now sorted based on a configured 'comparer' for a schema. You can configure the comparer by setting
+`ISchema.Comparer` to an implementation of `ISchemaComparer`. By default, introspection results are returned in the order they
+were defined.
+
+See [Default Sort Order of Introspection Query Results](#Default-Sort-Order-of-Introspection-Query-Results) below for a sample
+of how this can be used to return introspection results that are sorted alphabetically.
+
+### Array Pooling
+
+When returning lists of information from field resolvers, you can choose to rent an array from `IResolveFieldContext.ArrayPool`,
+populating it with your results and returning the array. The array will be released after the execution completes. This has
+limited uses, since the rented array is not guaranteed to be exactly the requested length, so the array would need to be
+wrapped in order to only return the correct number of entries, triggering a memory allocation (albeit a smaller one):
+```c#
+ resolve: context =>
+{
+    var ints = context.ArrayPool.Rent<int>(1000); // ints.Length >= 1000
+    for (int i=0; i<1000; ++i)
+        ints[i] = i;
+    return ints.Constrained(1000); // extension method to return an array or array-like object of a given length
+});
+It is not recommended to use this feature for interim calculations, as it is better to work directly with `System.Buffers.ArrayPool<T>`.
```

Comment:
```suggestion
```
It is not recommended to use this feature for interim calculations, as it is better to work directly with `System.Buffers.ArrayPool<T>`.
```